packages/circuit-ui/hooks/useCollapsible/useCollapsible.ts

```diff
@@ -76,18 +76,18 @@ export function useCollapsible<T extends HTMLElement = HTMLElement>({
       duration,
       onStart: () => {
         setHeight(getHeight(contentElement));
-        if (!isOpen) {
-          setOpen(true);
-        }
+        // The timeout forces the state update into the next animation frame.
```

Comment:
Oooooh a real-world application of [this kind of videos](https://www.youtube.com/watch?v=cCOL7MC4Pl0)

Reply:
I have to watch that now (maybe there's a better way).

Reply:
I've used `setImmediate` in the past but I believe it needs to be polyfilled in browsers.

Reply:
Actually now that you mention it, I'm remembering that maybe `requestAnimationFrame` would be better 🙈 never used it directly myself

Reply:
I saw this a couple years back, need a rewatch. Lunch and learn?

Reply:
I've watched the video and it has increased my admiration and appreciation for Jake Archibald to a whole other level 👏🏻👏🏻👏🏻

`setImmediate` is one of the rare features that's only supported by Internet Explorer and earlier versions of Edge (not the Webkit ones). The need for a polyfill makes this a non-starter for me.

`requestAnimationFrame` doesn't guarantee that the browser does a rendering pass between state updates, so it doesn't work for this use case either. 

`setTimeout` seems to be the best hack for this after all.

Reply:
Thanks for the recap! Let's ship this 🚢 

(@connor-baer. WDYT about a new minor? If there are any other changes we need to bring to the nav I think it's fine to do in patches—unless we have some breaking changes in mind?)

Reply:
@robinmetral Yes, I will prepare a v3.1 release today. Any changes to the new navigation components will (have to) be non-breaking.